### DISCUSSION TITLE
Handle promises in AssignmentNode

### DISCUSSION THREAD

**Discussion by iJhefe** [josdejong/mathjs#2948] [Q&A]
Can I make a Node like AssignmentNode handle promises ?

I have some async functions that are successfuly handled using `typed` and extending the builtin mathjs functions, but I don't know how can I handle them with **Node**s, like AssignmentNode.

i.e: 
```js
scope = new Map();
const value = await math.evaluate("sum(car(1), car(2)) * getTax()", scope) // Ok
// vs
const result = await math.evaluate(`
  tax = getTax()
  cars = sum(car(1), car(2))
  cars * tax
`, scope); // ResultSet with 3 entries Promise

const value = await result.entries.at(-1); 
console.log(value) // Final numeric result
console.log(scope.get('tax')) // Promise instead the resolved value (here I've wanted to be the numeric result)
console.log(await scope.get('tax')) // Numeric result ok
```
`car` and `getTax` are async functions and the result is calculated as well

**Comment by josdejong** [comment]
There is a topic about implementing async support, and there even is an old proof of concept. There where still some complex edge cases to be solved though.

Discussion: https://github.com/josdejong/mathjs/discussions/2665
POC: https://github.com/josdejong/mathjs/discussions/2665#discussioncomment-3414429
